src/charm.py

```diff
@@ -948,13 +1080,59 @@ def _generate_vault_config_file(self) -> None:
             node_id=self._node_id,
             retry_joins=retry_joins,
         )
+        content = self._add_transit_stanza_if_needed(content)
         existing_content = ""
         if self._container.exists(path=VAULT_CONFIG_FILE_PATH):
             existing_content_stringio = self._container.pull(path=VAULT_CONFIG_FILE_PATH)
             existing_content = existing_content_stringio.read()
 
         if not config_file_content_matches(existing_content=existing_content, new_content=content):
             self._push_config_file_to_workload(content=content)
+            if _contains_transit_stanza(existing_content) != _contains_transit_stanza(content):
+                if self._vault_service_is_running():
+                    self._container.restart(self._service_name)
+
+    def _add_transit_stanza_if_needed(self, config_content: str) -> str:
```

Comment:
I feel like this function is doing a bit more than just adding new config options to the config content it receives, it basically is doing all the configuration that is needed for the autounseal requirer, for example pushing the CA cert and authenticating agains the provider vault.

Reply:
That's a fair point. It could be refactored to be a ~~little~~ lot cleaner.

Reply:
Fixed in https://github.com/canonical/vault-k8s-operator/pull/365/commits/0a90837bbcdef6e42799e4e45129a758b899699f